sql/hive-thriftserver/v2.3/src/main/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java

```diff
@@ -144,6 +144,9 @@ public void run() {
       // TODO: check defaults: maxTimeout, keepalive, maxBodySize, bodyRecieveDuration, etc.
       // Finally, start the server
       httpServer.start();
+      // In case HIVE_SERVER2_THRIFT_HTTP_PORT or hive.server2.thrift.http.port is configured with
+      // 0 which represents any free port, we should set it to the actual one
+      portNum = connector.getLocalPort();
```

Comment:
@yaooqinn did you try to test that? For me setting hive.server2.thrift.http.port to 0 doesn't seem to work when trying to start it in http mode like that.

Reply:
```
 kentyao@hulk  ~/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-20200601  sbin/start-thriftserver.sh --conf spark.hadoop.hive.server2.thrift.http.port=0 --conf spark.hadoop.hive.server2.transport.mode=http
starting org.apache.spark.sql.hive.thriftserver.HiveThriftServer2, logging to /Users/kentyao/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-20200601/logs/spark-kentyao-org.apache.spark.sql.hive.thriftserver.HiveThriftServer2-1-hulk.local.out
 kentyao@hulk  ~/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-20200601 tail -f -2 /Users/kentyao/Downloads/spark/spark-3.1.0-SNAPSHOT-bin-20200601/logs/spark-kentyao-org.apache.spark.sql.hive.thriftserver.HiveThriftServer2-1-hulk.local.out
20/06/05 21:26:20 INFO HiveThriftServer2: HiveThriftServer2 started
20/06/05 21:26:20 INFO ThriftCLIService: Started ThriftHttpCLIService in http mode on port 54379 path=/cliservice/* with 5...500 worker threads
```

Reply:
Thanks... let me debug this more on my end why am I getting 0 back...

Reply:
I verify this locally, it seems to be ok.

but in https://github.com/apache/spark/pull/28738, I just run into the same issue.